### PR TITLE
HPCC-11482 Automatically add manifests named same as included ECL files

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@GordonSmith Please review

File: ecl/eclcc/eclcc.cpp [hunk 1]
```diff
@@ -714,6 +719,26 @@ void EclCC::reportCompileErrors(IErrorReceiver & errorProcessor, const char * pr
...
+    if (!tree)
+        return;
+    Owned<IPropertyTreeIterator> iter = tree->getElements((instance.archive) ? "Module/Attribute" : "Attribute");
+    ForEach(*iter)
+    {
+        StringBuffer filename(iter->query().queryProp("@sourcePath"));
+        if (filename.length())
+        {
+            getFullFileName(filename, true).append(".manifest");
+            if (filenames.contains(filename))
```
Comment (GordonSmith) [inline ecl/eclcc/eclcc.cpp]: Can the same manifest get added twice, if it was also on the command line?  (The IDE will also automatically add a manifest file to he command line arguments if it matches the file being submitted).

**Comment by GordonSmith** [comment]
Looks ok - just a minor query on what happens if the IDE passes in a manifest which it will then also auto-discover.

**Comment by richardkchapman** [comment]
@afishbeck , over to you

File: ecl/eclcc/eclcc.cpp (hunk 1 above)
Comment (afishbeck) [inline ecl/eclcc/eclcc.cpp]: @GordonSmith the command line manifest file name will have already been added to this list (line 500), therefore filenames.contains() would find it.

**Comment by afishbeck** [comment]
@richardkchapman responded in line.

**Pull request merged**